src/Core/Canvas.cpp

```
@@ -303,21 +303,24 @@ void Canvas::box(float x1, float y1, float x2, float y2, float radius) {
 }
 
 void Canvas::box(int x1, int y1, int x2, int y2, float radius) {
-    box((float)x1, (float)y1, (float)x2, (float)y2, radius);
+    box(static_cast<float>(x1), static_cast<float>(y1), static_cast<float>(x2),
+        static_cast<float>(y2), radius);
 }
 
 void Canvas::polygon(const float* x, const float* y, int vertexCount) {
     if (vertexCount < 3) return;
     GetPolyDist func(x, y, vertexCount);
     data_->draw(canvas_data_, canvas_width_, canvas_height_,
-                {0, 0, (float)canvas_width_, (float)canvas_height_}, &func);
+                {0, 0, static_cast<float>(canvas_width_),
+                 static_cast<float>(canvas_height_)},
+                &func);
 }
 
 Texture Canvas::createTexture(bool mipmap) const {
-    uint8_t* dst =
```

not for this PR, future - addressing memory safety things?

---

Yeah. There are a lot of mallocs/reallocs that don't check for error conditions.
I changed the ones that threw warnings to fail silently, but we should add exception handling and all that.